lib/controllers/id-site-verify.js

```diff
@@ -18,31 +21,78 @@ module.exports = function (req, res) {
   var config = req.app.get('stormpathConfig');
   var logger = req.app.get('stormpathLogger');
 
-  application.handleIdSiteCallback(req.originalUrl, function (err, result) {
+  var params = req.query || {};
+  var stormpathToken = params.jwtResponse || '';
+  var assertionAuthenticator = new stormpath.StormpathAssertionAuthenticator(application);
+
+  assertionAuthenticator.authenticate(stormpathToken, function (err) {
     if (err) {
-      logger.info('While attempting to authenticate a user via ID site, the callback verification failed.');
-      return res.status(500).end(err.toString());
+      logger.info('During an IdSite login attempt, we were unable to verify the JWT response.');
+      return res.status(err.status || 400).json(err);
     }
 
-    var idSiteNextUri = config.web.idSite.nextUri;
+    var parsedToken = nJwt.verify(stormpathToken, config.client.apiKey.secret);
+    var tokenStatus = parsedToken.body.status;
 
-    switch (result.status) {
-      case 'LOGOUT':
-        middleware.deleteCookies(req, res);
-        res.redirect(302, idSiteNextUri);
-        break;
+    function redirectNext() {
+      var nextUri = config.web.idSite.nextUri;
+      var nextQueryPath = url.parse(params.next || '').path;
+      res.redirect(302, nextQueryPath || nextUri);
+    }
+
+    function authenticateToken(callback) {
+      var stormpathTokenAuthenticator = new stormpath.OAuthStormpathTokenAuthenticator(application);
+
+      stormpathTokenAuthenticator.authenticate({ stormpath_token: stormpathToken }, function (err, authenticationResult) {
+        if (err) {
+          logger.info('During an IdSite login attempt, we were unable to create a Stormpath session.');
+          return res.status(err.status || 400).json(err);
+        }
+
+        helpers.createSession(authenticationResult, authenticationResult.account, req, res);
+
+        callback();
+      });
+    }
+
+    function handleAuthRequest(type, callback) {
+      var handler = config['post' + type + 'Handler'];
```

Comment:
I generally avoid dynamic function names, as it makes the code harder to read (IMO), and I'm pretty sure it makes the job difficult for static analyzers.